daemon.go

```diff
@@ -59,12 +60,27 @@ func (daemon menderDaemon) StopDaemon() {
 
 func (daemon *menderDaemon) Run() error {
 	// figure out the state
-	switch daemon.GetState() {
-	case MenderFreshInstall:
-		//do nothing
-	case MenderRunningWithFreshUpdate:
-		daemon.CommitUpdate()
+	for {
+		switch daemon.TransitionState() {
+		case MenderStateRunningWithFreshUpdate:
+			daemon.CommitUpdate()
+		case MenderStateWaitForUpdate:
+			err := daemon.waitForUpdate()
+			if err != nil {
+				return err
```

Comment:
If for some reason, you wont be able to fetch update, you will kill the daemon, which is not what I would expect.